acoustic_msgs/Contributing.md

```diff
@@ -0,0 +1,133 @@
+   closed. Release branches will be tested at sea (possibly for several
+   deployments) and may include more than one feature
+5. Push further fixes to the release branch
+6. Finally, merge the release branch to Master via pull request
+
+Example:
+
+            git checkout main
+            git pull --rebase
+            git checkout -b feature/2019-04-29-cool-new-feature
+            # develop on your feature branch...
+
+            *--A---B----------------------C--  main
+                    \                    /
+                     \            *--R--S    release
+                      \             /
+                       N---O---P---Q    cool feature
+
+            # done? create PR to release and schedule a CR
+
+Don't forget to keep your feature branch up to date with main!
+
+Say, you have:
+
+            *--A---B---C---D---E--- main
+                \
+                 N---O---P---Q----- feature
+
+            # run
+            git fetch origin # update origin/main from the server
+            git stash # if you have uncommitted local changes
+
+            # then do
+            git checkout main # check out your local tracking branch ...
+            git pull --rebase # ... and bring it up to date
+            git checkout feature/2019-04-29-cool-new-feature # go back to your feature branch
+            git merge master # do the actual merge to obtain:
+
+             *--A---B---C---D---E--- main
+                 \               \
+                  N---O---P---Q---R-- feature
+
+            # after merging, don't forget to
+            git stash pop
+
+### Fixing/updating existing code
+
+1. Create a bugfix/hotfix branch and commit your changes to it. Each fix should have a dedicated branch
+2. Test in sim / lab / test tank / at sea!
+3. Create a pull request to Master
+4. Propagate changes to release and feature branches
+
+Example:
+
+            git checkout main
+            git pull --rebase
+            git checkout -b bugfix/2019-04-29-fix-nasty-bug
+            # commit changes to your fix branch...
+
+             *--A---B-----C--  main
+                     \    /\
+              bugfix  X--Y  \
+                             \
+                  N---O---P---Q some release
+
+            # done? create a PR to main and merge to release/feature
+
+Note about hotfixes: please consider creating a feature branch depending on the scope of the changes you're introducing.
+
+## Code of conduct
+
+1. Never commit code that brakes the build
+
+        # also, please test your changes by running the project's unit/regression tests
+        cd path/to/hydrographic_msgs
+
+        # run
+        ??? # to execute unit tests and ...
+        ??? # ... for regression tests
+
+2. Make sure your code meets the project's style guidelines before committing
+
+        # keeping your code in style is easy!
+        make pretty # to format the code (ansi, aStyle 2.05.1) and ...
+        make cxxcheck # ... to check Cxx style
```

Comment:
I like the idea of using automatic code formatting in general but I think we could use a little more details on this.  Is this something we need to set up in our cmake files?

Reply:
I stripped this section in my proposed revision because I didn't find a message-specific style guide in a quick search:
* Do you know of a standard tool for formatting .msg files? If not, here's a possible start:
  * Comments on line(s) *above* field definition (not carefully formatted in 2nd column)
  * Line length 80
  * No trailing spaces
  * message names underscored
* Do you already have tooling that you can suggest for that? (I *love* having auto code formatting, but haven't set it up in any of my current projects)

Reply:
For auto formatting I use the beautifier plugin for qt-creator.  I believe it just calls clang code formatting.   My method isn't super portable or built into the project itself though.  Maybe @braanan  has a suggestion?

This might be a mute point anyway if there is no formatter for msg files.   I don't know of any.  

@lauralindzey I think you mean message fields are underscored.   Message names (types) are camel cased.  http://wiki.ros.org/ROS/Patterns/Conventions#Messages

Reply:
Yup, whoops! Meant field names not message names. I'll edit the above comment to be right.